### ISSUE TITLE
Feature Request: Add/Override additional environments for Google services files

### ISSUE THREAD

**Issue by ScottAtRedHawk** [EddyVerbruggen/nativescript-plugin-firebase#1515]
Hi Eddy,
I think it'd be handy to have a way to add additional Firebase environments, or override the defaults. Currently the plugin supports development and production environments, but other ones like staging or pre-production would be good too. I don't know if it's possible but a good place to control these might be in the firebase.nativescript.json file as an array of the the specified environments. These would correspond to the suffixes on the google-services.json and GoogleServices-Info.plist files.

Example
--------
- `prod` will be selected if you run with either the `--release`, `--env.prod` or `--env.production` flags
- `staging` will be selected if you run with `--env.staging` (the environment string must match what is in the configuration array, filename and command-line arguments). If you combine `--release` with `--env.staging`. then `staging` would be selected and the app would be built as a release version.
- `dev` will be selected if you do not run with any of the above flags

```
// in firebase.nativescript.json file
{
  ...
  "firebase_environments": [ "dev", "staging", "prod" ]
  ...
}

// above would map to:
// Android
//    google-services.json.dev
//    google-services.json.staging
//    google-services.json.prod

// iOS
//    GoogleServices-Info.plist.dev
//    GoogleServices-Info.plist.staging
//    GoogleServices-Info.plist.prod
```

**Comment by EddyVerbruggen** [comment]
Hi,

I wouldn't mind having support for it in the plugin, but just so you know: I won't be adding it, although I'd be happy to merge a PR from the community.

**Comment by allangoncalves** [comment]
It would be great if I could put it into the project I'm currently working.

If not, I would love to allow production only with an explicit tag. I'm currently using --release for staging builds and I cannot set to dev automatically.

**Comment by johan-cuello-cb** [comment]
Hi @EddyVerbruggen ,

Is it possible to connect to multiple firebase apps, like mentioned in the Firebase documenttion https://firebase.google.com/docs/projects/multiprojects?

And if you think is possible where can I start looking at?

Regards

**Comment by manojdcoder** [comment]
I think should be possible, may have to just tweak the way the instance is accessed now.

**Comment by johan-cuello-cb** [comment]
@manojdcoder , you are right. I think a Factory pattern could be used to get instances based on configuration file, but how can I read these files? Which class is it used to read the configuration file?

**Comment by manojdcoder** [comment]
@johan-cuello-cb Right now the plugin supports just single project and technically the compiler / SDK itself takes care of reading the credentials. I imagine for configuring multiple projects, you will just directly pass the credentials to the SDK as shown in the docs.
